File: c2-chaincode/chaincode-api/chaincode-api-fabric/src/main/java/io/komune/c2/chaincode/api/fabric/utils/JsonUtils.java
package io.komune.c2.chaincode.api.fabric.utils;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import java.io.IOException;
import java.io.InputStream;
import java.net.URL;
import tools.jackson.databind.json.JsonMapper;

public class JsonUtils {

    private static final JsonMapper mapper;

    static {
        mapper = JsonMapper.builder()
                .changeDefaultVisibility(vc -> vc
                        .withFieldVisibility(JsonAutoDetect.Visibility.ANY))
                .build();
    }

    public static String toJson(Object obj) {
        return mapper.writeValueAsString(obj);
    }

    public static <T> T toObject(URL url, Class<T> clazz) throws IOException {
        try (InputStream is = url.openStream()) {
            return mapper.readValue(is, clazz);
        }
    }

}

File: c2-chaincode/chaincode-api/chaincode-api-config/src/main/kotlin/io/komune/c2/chaincode/api/config/utils/JsonUtils.kt
package io.komune.c2.chaincode.api.config.utils

import java.io.IOException
import java.net.URL
import tools.jackson.databind.ObjectMapper
import tools.jackson.databind.json.JsonMapper
import tools.jackson.module.kotlin.KotlinModule
import tools.jackson.module.kotlin.readValue


object JsonUtils {
    val mapper: ObjectMapper = JsonMapper.builder()
        .addModule(KotlinModule.Builder().build())
        .build()

    fun toJson(obj: Any?): String {
        return mapper.writeValueAsString(obj)
    }

    @Throws(IOException::class)
    fun <T> toObject(value: URL?, clazz: Class<T>?): T {
        return mapper.readValue(value!!.openStream(), clazz)
    }
    @Throws(IOException::class)
    inline fun <reified T> toObject(value: URL): T {
        return mapper.readValue(value.openStream())
    }
}
